common/lib/xmodule/xmodule/modulestore/xml_exporter.py

```diff
@@ -42,6 +42,9 @@ def _export_drafts(modulestore, course_key, export_fs, xml_centric_course_key):
             qualifiers={'category': {'$nin': DIRECT_ONLY_CATEGORIES}},
             revision=ModuleStoreEnum.RevisionOption.draft_only
         )
+        # Check to see if the returned draft modules have changes w.r.t. the published module.
+        # Only modules with changes will be exported into the /drafts directory.
+        draft_modules = [module for module in draft_modules if modulestore.has_changes(module)]
```

Comment:
It seems like, given the `revision=ModuleStoreEnum.RevisionOption.draft_only` qualifier on line 43, any modulestore that returns non-draft content (content where has_changes is false) is misbehaving. Although, I guess Studio allows for drafts that don't differ in content from the published version (that's what you get when you first hit "Edit draft" in Studio). Blarg. Such unclear semantics.